astropy/io/fits/tests/test_header.py

```diff
@@ -3211,3 +3211,70 @@ def append(self, card, *args, **kwargs):
     assert copy_.comments["c"] == "no comment"
     my_header.extend((("d", 4.0),))
     assert my_header.comments["d"] == "no comment"
+
+
+class HeaderWithValidation(fits.Header):
+    @classmethod
+    def _validate_card(cls, card):
+        if card.keyword.startswith("B") or card.keyword == "ABC":
+            raise ValueError("Keyword cannot start with a 'B' or be 'ABC'")
+
+        if isinstance(card.value, (int, float)) and card.value < 0:
+            warnings.warn("Value cannot be negative, ignoring", UserWarning)
+            return None
+
+        if card.comment is not None and card.comment.lower() != card.comment:
+            warnings.warn("Comment should be lowercase, fixing", UserWarning)
+            card.comment = card.comment.lower()
+            return card
```

Comment:
Here, `return card` should not be indented (or there should be another clause with a `return` at the end).

Reply:
Done